ms-frontend/src/components/CreateReviewModal.vue

```diff
@@ -0,0 +1,226 @@
+import { computed, ref, watch } from 'vue'
+import { useAuthStore } from '@/stores/auth'
+import FileUpload from '@/components/FileUpload.vue'
+import { reviewService } from '@/services/reviewService'
+import Textarea from 'primevue/textarea'
+import type { User } from '@/types/User'
+import type { Location } from '@types/Location'
+import type { Review } from '@types/Review'
+
+const authStore = useAuthStore()
+
+const signedInUser = authStore.user
+const myId = signedInUser?.id ?? ''
+
+const rating = ref<number>(0)
+const reviewText = ref<string>('')
+const ratingValid = ref<boolean>(false)
+const textValid = ref<boolean>(false)
+
+const props = defineProps<{
+    location: Location
+}>()
+
+
+
+const emit = defineEmits(['closeModal'])
+
+const reviewPicture = ref<File | null>(null)
+
+watch(reviewText, (newValue) => {
+    const reviewTextWords = newValue.trim().split(/\s+/).length
+    const reviewTextValue = newValue.trim()
+    if (reviewTextWords > 5 && reviewTextWords < 200 && reviewTextValue !== '') {
+        textValid.value = true
+    } else {
+        textValid.value = false
+    }
+})
+
+watch(rating, (newValue) => {
+    if (rating.value > 0) {
+        ratingValid.value = true
+    } else {
+        ratingValid.value = false
+    }
+})
+
+const submitRating = (selectedRating: number) => {
+    rating.value = selectedRating;
+}
+
+const checkInputValidity = () => {
+    const reviewTextWords = reviewText.value.trim().split(/\s+/).length;
+    const reviewTextValue = reviewText.value.trim();
+
+    if (reviewTextWords > 5 && reviewTextWords < 200 && reviewTextValue !== '') {
+        textValid.value = true
+    }
+    if (rating.value > 0) ratingValid.value = true
+    return ratingValid.value && textValid.value
+}
+
+// const testReviews = async () => {
+//     const myreviews = ref<Review[]>([])
+//   try{
+//     myreviews.value = await reviewService.getUserReviews(myId)
+//     console.log(myreviews.value.reviews[0])
+//   }
+//   catch(error){
+//     console.log('Error testing myreviews:', error)
+//   }
+// }
+
+const createReview = async () => {
+    if (checkInputValidity()) {
+        try {
+            // Call createReview method from reviewService
+            await reviewService.createReview(reviewText.value, rating.value, props.location)
+            emit('closeModal')
+            
+        } catch (error) {
+            console.error('Error creating review:', error)
+        }
+    }
+}
+
+// const updateReviewImage = async () => {
+
+//     const myReviews = ref<Review[]>()
+//     try {
+//         myReviews.value = await reviewService.getUserReviews(myId)
+//     } catch (error) {
+//         console.error('Error accessing the created reviews:', error)
+//     }
+
+//     try {
+//         await reviewService.updateReviewImage(myReviews.value?.[0].review_id, reviewPicture)
+//     } catch (error) {
+//         console.error('Error updating review image:', error)
+//     }
+// }
```

Comment:
```suggestion
const createReview = async () => {
  if (checkInputValidity()) {
    try {
      // Call createReview method from reviewService
      await reviewService.createReview(reviewText.value, rating.value, props.location).then((response: Review) => {
        updateReviewImage(response.id)
      })
      emit('closeModal')
    } catch (error) {
      console.error('Error creating review:', error)
    }
  }
}

const updateReviewImage = async (reviewId: string) => {
    if(reviewPicture.value === null) return
    try {
        await reviewService.appendReviewImage(reviewId, reviewPicture.value)
    } catch (error) {
        console.error('Error updating review image:', error)
    }
}
```

With this you could update the profile picture after the review was created.

Reply:
This gives me 'Request failed with status code 400', trying to look at it now

Reply:
Did you add this [Change](https://github.com/uzh-network-ninjas/ase-social-network/pull/153/files#r1582062079) ?

Reply:
yup, still gives the 400 bad request error:/